libs/ml/include/ml/optimisation/sgd_optimiser.hpp

```diff
@@ -57,6 +58,9 @@ class SGDOptimiser : public Optimiser<T>
   }
 
 private:
+  // ApplyGradientSparse if number_of_rows_to_update*sparsity_threshold_ <= total_rows
+  SizeType sparsity_threshold_ = 4;
```

Comment:
it would be nice to have more explanation about this. Is this number empirically derived from benchmarking results? If so it would be good to point that out and state which benchmark illustrates this